src/extrinsic/mod.rs

```diff
@@ -41,6 +44,9 @@ use crate::{
     Error,
 };
 
+/// A reasonable default for `mortal_period`
+pub const DEFAULT_MORTAL_PERIOD: u64 = 64;
```

Comment:
If I understand this correctly, this would works as long `BlockHashCount` >= `64` but we can't know for use because it's defined by the runtime. Thus, I prefer to fetch `block_hash_count` from the runtime and use it as `default_mortal_period` (could be something within the range 1..=block_hash_count)

Reply:
I think your proposed approach is ideal, but after looking into it a little bit it seems like this is not possible without implementing constants for metadata. I think we would need support for this: https://docs.rs/frame-metadata/11.0.0-alpha.3/frame_metadata/struct.ModuleMetadata.html#structfield.constants. You can even see here we currently have a place holder: https://github.com/paritytech/substrate-subxt/blob/8a76bebcff6534421b498ac874e7e6ffe09d256a/src/metadata.rs#L169 ... I could be wrong though, maybe there is a way to just key into storage?

Reply:
https://github.com/paritytech/substrate-subxt/issues/206

Also realizing though right after making that issue, it is ok to have a `mortal_period` that is too high. In that case the extrinsic mortality will simply end at `BlockHashCount`.

Reply:
all right, I see